src/query/plan/preprocess.cpp

```diff
@@ -623,4 +633,19 @@ QueryParts CollectQueryParts(SymbolTable &symbol_table, AstStorage &storage, Cyp
   return QueryParts{query_parts, distinct};
 }
 
+// NOTE: matchings is defined here because it can't be defined in hpp because of forward declare -> take care if you add
+// another constructor.
+FilterInfo::FilterInfo() : matchings{} {}
+FilterInfo::~FilterInfo() = default;
+
+FilterInfo::FilterInfo(const FilterInfo &) = default;
+FilterInfo &FilterInfo::operator=(const FilterInfo &) = default;
+FilterInfo::FilterInfo(FilterInfo &&) noexcept = default;
+FilterInfo &FilterInfo::operator=(FilterInfo &&) noexcept = default;
```

Comment:
Can this also be moved to hpp?

Reply:
I'm pretty sure it's not possible because `FilterInfo` is not fully defined under hpp file (take a look at the comment under hpp).